src/LocalTranslationsBatcher.php

```diff
@@ -113,18 +115,29 @@ public function createBatch(array $langcodes = []): void {
     }
 
     $extensions = $this->getExtensionsToTranslate();
-    if (!$extensions) {
+    if (!$extensions || !$langcodes) {
       return;
     }
 
-    locale_translation_flush_projects();
-    locale_translation_check_projects_local($extensions, $langcodes);
-    $options = _locale_translation_default_update_options();
-    $batch = locale_translation_batch_fetch_build($extensions, $langcodes, $options);
-    batch_set($batch);
-    if ($batch = locale_config_batch_update_components($options, $langcodes)) {
-      batch_set($batch);
+    // Build operations for local translation batch import.
+    $operations = [];
+    foreach ($extensions as $extension) {
+      $operations[] = [
+        [get_class($this), 'importProjectPoFiles'],
```

Comment:
Can you not pass the entire object as a callable? `[$this, 'importProjectPoFiles']`? This way when it's called it would be instantiated with dependencies and we can inject `\Drupal::service('file_system')`

Also, we are adding an operation for each extension. Instead we should only have one operation within which we use the &$context to perform the job on chunks of extensions and keep track of them all.

Reply:
Done! Thank you for tip.